### PR TITLE
Run CrossBuild as agent user to avoid creating undeletable files

### PR DESCRIPTION
**Pull request by dagood**
Create `docker-as-current-user.sh` for both `run` and `exec`. Use this script in the CrossBuild so that files created in the mapped volume are owned by the agent user. This allows VSTS to clean up.

Since the user isn't root anymore, I had to set `HOME` to something other than `/`. `export`ing the agent account's home into the container results in a `HOME` that doesn't exist, so I added that as another condition to use the custom `HOME` code in build.sh. (Trying to simply clear `HOME` to activate the custom code results in `/` being set, and I didn't want to use the workaround of sending a random garbage value.)

I also moved the mapped container git dir from `/root/...` to `/opt/code/...` for permissions.

@MichaelSimons @gkhanna79 PTAL

I noticed when I was testing that this def doesn't have any pull-retry logic, so it failed to pull the image quite often in my test runs. Created https://github.com/dotnet/core-setup/issues/1816 for this, but it might be tracked somewhere else, too. /cc @ravimeda

### CODE REVIEW

File: buildpipeline/Core-Setup-CrossBuild.json [hunk 1]
```diff
@@ -61,32 +61,34 @@
       "displayName": "Start detached container",
```
Comment (MichaelSimons) [inline buildpipeline/Core-Setup-CrossBuild.json]: This is not directly related to your change, but I'm not seeing why we have 4 docker steps - start container, build in container, stop container, remove container.  Seems like this could be accomplished in one step.

Comment (dagood) [inline buildpipeline/Core-Setup-CrossBuild.json]: Agreed. IIRC this is because this def was copied from CoreFX and tweaked for Core-Setup.

**Comment by dagood** [comment]
@dotnet-bot test Fedora23 x64 Debug Build
